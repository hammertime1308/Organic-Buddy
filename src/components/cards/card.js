import React from 'react';
import { Image } from 'react-native';

import { NavigationService } from '../../utilities';

import { StyledCard, Title, Description, Price } from './styles';

export const Card = ({
  id = '',
  productName = 'Title',
  productDescription = 'Description',
  productPrice = 0,
  productImages = [],
}) => {
  return (
    <StyledCard
      onPress={() =>
        NavigationService.navigate('DetailedScreenFS', {
          id,
          productName,
          productDescription,
          productPrice,
          productImages,
        })
      }
      activeOpacity={0.75}>
      {productImages.length !== 0 ? (
        <Image
          style={{
            width: '100%',
            height: 120,
            resizeMode: 'cover',
          }}
          source={{
            uri: `${productImages[0]}`,
          }}
        />
      ) : null}
      <Title>{productName}</Title>
      <Description>{productDescription}</Description>
      <Price>Price: {productPrice} Rs</Price>
    </StyledCard>
  );
};
