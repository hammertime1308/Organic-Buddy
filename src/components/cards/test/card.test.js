import React from 'react';
import renderer from 'react-test-renderer';
import { Card } from '../card';

describe('Card', () => {
  it('Should render card without image', () => {
    const tree = renderer
      .create(
        <Card
          id="1"
          productName="Sample name for test"
          productDescription="sample description of the card"
          productPrice={10200}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render card with image', () => {
    const tree = renderer
      .create(
        <Card
          id="1"
          productName="Sample name for test"
          productDescription="sample description of the card"
          productPrice={10200}
          productImages={[
            'https://cdn.pixabay.com/photo/2017/07/19/09/51/hallstatt-2518544_960_720.jpg',
          ]}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
