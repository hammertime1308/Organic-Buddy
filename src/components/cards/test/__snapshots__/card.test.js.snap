// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render card with image 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffff",
      "borderColor": "rgba(0, 0, 0, 0.01)",
      "borderRadius": 5,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "elevation": 3,
      "flexDirection": "column",
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOpacity": 0.26,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://cdn.pixabay.com/photo/2017/07/19/09/51/hallstatt-2518544_960_720.jpg",
      }
    }
    style={
      Object {
        "height": 120,
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "marginBottom": 5,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 10,
        },
      ]
    }
  >
    Sample name for test
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 15,
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 0,
        },
      ]
    }
  >
    sample description of the card
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 15,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 0,
        },
      ]
    }
  >
    Price: 
    10200
     Rs
  </Text>
</View>
`;

exports[`Card Should render card without image 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffff",
      "borderColor": "rgba(0, 0, 0, 0.01)",
      "borderRadius": 5,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "elevation": 3,
      "flexDirection": "column",
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOpacity": 0.26,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "marginBottom": 5,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 10,
        },
      ]
    }
  >
    Sample name for test
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 15,
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 0,
        },
      ]
    }
  >
    sample description of the card
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 15,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 0,
        },
      ]
    }
  >
    Price: 
    10200
     Rs
  </Text>
</View>
`;
